### PR TITLE
Create lobby

### PR DESCRIPTION
**Pull request by matiascalegaris**
Create player lobby module to be used by scripted events
Add a generic lobby to be used by GM for basic events

### CODE REVIEW

File: Codigo/ModLobby.bas [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            UserIndex = .UserId
+            If Not .IsSummoned Then
+                .SummonedFrom = UserList(UserIndex).Pos
+            End If
+100         Call WarpToLegalPos(UserIndex, instance.SummonCoordinates.map, instance.SummonCoordinates.X, instance.SummonCoordinates.y, True, True)
+            .IsSummoned = True
+        End With
+    Exit Sub
+SummonPlayer_Err:
+102     Call TraceError(Err.Number, Err.Description, "ModLobby.SummonPlayer_Err", Erl)
```
Comment (morgolock) [inline Codigo/ModLobby.bas:144]: 102 and 100 are repeated in the same file, this should give compiler errors. Please remove line numbers.

**Review by morgolock — CHANGES_REQUESTED** [review]

File: Codigo/ModLobby.bas (hunk 1 above)
Comment (matiascalegaris) [reply Codigo/ModLobby.bas:144]: these line numbers are on different functions, this builds fine, you can see more examples of this in protocol.bas

File: Codigo/Protocol.bas [hunk 2]
```diff
@@ -10719,20 +10718,159 @@ ErrHandler:
...
+                Value = Reader.ReadInt32()
+                If hasPermission Then
+                    Call ModLobby.SummonPlayer(GenericGlobalLobby, Value)
+                    Exit Sub
+                End If
+            Case e_LobbyCommandId.eListPlayers
+                If hasPermission Then
+                    Call ModLobby.ListPlayers(GenericGlobalLobby, UserIndex)
+                    Exit Sub
+                End If
```
Comment (morgolock) [inline Codigo/Protocol.bas:10791]: please add 

```
case else
exit sub
```

File: Codigo/Protocol.bas [hunk 3]
```diff
@@ -10719,20 +10718,159 @@ ErrHandler:
...
+Private Sub HandleStartEvent(ByVal UserIndex As Integer)
+On Error GoTo ErrHandler
+    Dim EventType As Byte
+    EventType = Reader.ReadInt8()
+    
+    Select Case EventType
+        Case e_EventType.CaptureTheFlag
+            HandleIniciarCaptura (UserIndex)
+        Case e_EventType.Generic
+            HandleStartGenericLobby (UserIndex)
```
Comment (morgolock) [inline Codigo/Protocol.bas:10813]: missing `case else exit sub`

File: Server.VBP [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
-Reference=*\G{420B2830-E718-11CF-893D-00A0C9054228}#1.0#0#..\..\..\..\..\Windows\SysWOW64\scrrun.dll#Microsoft Scripting Runtime
-Reference=*\G{F5078F18-C551-11D3-89B9-0000F81FE221}#6.0#0#..\..\..\..\..\Windows\SysWOW64\msxml6.dll#Microsoft XML, v6.0
-Reference=*\G{88D89F7C-BB59-4F13-9850-93CBD874039D}#1.5#0#..\..\..\..\..\Windows\SysWow64\Aurora.Network.dll#Aurora.Network - Aurora Engine - Network
-Object={6B7E6392-850A-101B-AFC0-4210102A8DA7}#1.3#0; COMCTL32.OCX
+Reference=*\G{B691E011-1797-432E-907A-4D8C69339129}#6.1#0#..\..\Program Files (x86)\Common Files\System\ado\msado15.dll#Microsoft ActiveX Data Objects 6.1 Library
+Reference=*\G{420B2830-E718-11CF-893D-00A0C9054228}#1.0#0#..\..\Windows\SysWOW64\scrrun.dll#Microsoft Scripting Runtime
+Reference=*\G{F5078F18-C551-11D3-89B9-0000F81FE221}#6.0#0#..\..\Windows\SysWOW64\msxml6.dll#Microsoft XML, v6.0
+Reference=*\G{88D89F7C-BB59-4F13-9850-93CBD874039D}#1.5#0#Aurora.Network.dll#Aurora.Network - Aurora Engine - Network
+Object={6B7E6392-850A-101B-AFC0-4210102A8DA7}#1.3#0; comctl32.ocx
```
Comment (morgolock) [inline Server.VBP:6]: please undo these changes are not necessary

we just need to add the line 90

**Review by morgolock — CHANGES_REQUESTED** [review]

File: Codigo/ModLobby.bas [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+ReturnAllPlayer_Err:
+102     Call TraceError(Err.Number, Err.Description, "ModLobby.ReturnAllPlayer", Erl)
+End Sub
+
+Public Sub CancelLobby(ByRef instance As t_Lobby)
+On Error GoTo CancelLobby_Err
+    Call ReturnAllPlayers(instance)
+    instance.RegisteredPlayers = 0
+    instance.State = Closed
+CancelLobby_Err:
```
Comment (morgolock) [inline Codigo/ModLobby.bas:190]: missing `exit sub`

File: Codigo/Protocol.bas [hunk 6]
```diff
@@ -10719,20 +10718,163 @@ ErrHandler:
...
+                Call WriteConsoleMsg(UserIndex, "El nivel minimo debe ser menor al maximo.", e_FontTypeNames.FONTTYPE_INFO)
+                Exit Sub
+            End If
+            Call InitializeLobby(GenericGlobalLobby)
+            Call ModLobby.SetMinLevel(GenericGlobalLobby, minLevel)
+            Call ModLobby.SetMaxLevel(GenericGlobalLobby, maxLevel)
+            Call ModLobby.SetMaxPlayers(GenericGlobalLobby, maxPlayers)
+            Call WriteConsoleMsg(UserIndex, "Se creo el lobby, recorda que tenes que abrirlo para que se pueda anotar gente.", e_FontTypeNames.FONTTYPE_INFO)
+        End If
+    End With
```
Comment (morgolock) [inline Codigo/Protocol.bas:10858]: `exit sub` after `end with`

**Review by morgolock — CHANGES_REQUESTED** [review]

**Review by morgolock — APPROVED** [review]
